saaj-ri/src/main/java/com/sun/xml/messaging/saaj/util/stax/SaajStaxWriter.java

```diff
@@ -489,13 +491,20 @@ public SOAPElement flushTo(final SOAPElement target) throws XMLStreamException {
                         newElement = target.addChildElement(this.localName, this.prefix, this.namespaceUri);
                     }
                     // add namespace declarations
-                    for (NamespaceDeclaration namespace : this.namespaceDeclarations) {
-                        newElement.addNamespaceDeclaration(namespace.prefix, namespace.namespaceUri);
+                    for (Map.Entry<String, String> namespace : this.namespaceDeclarations.entrySet()) {
+                        newElement.addNamespaceDeclaration(namespace.getValue(), namespace.getKey());
                     }
                     // add attribute declarations
                     for (AttributeDeclaration attribute : this.attributeDeclarations) {
+                        String pfx = attribute.prefix;
+                        if ("".equals(emptyIfNull(pfx))) {
```

Comment:
Isn't this emptyIfNull(pfx) call redundant since the only place where we are adding to the attributeDeclarations is line 460 and prefix is already handled this way there. You can also use just regular pfx.isEmpty() now.